File: src/main/java/com/jetbrains/shared/FileStorage/FileStorageService.java
package com.jetbrains.shared.FileStorage;

import org.springframework.core.io.Resource;
import org.springframework.web.multipart.MultipartFile;

public interface FileStorageService {

    String storeFile(MultipartFile file);

    Resource downloadFile(String fileName);

    void deleteFile(String fileName);

    String getFileExtension(String fileName);

}

File: src/main/java/com/jetbrains/shared/FileStorage/FileStorageService.kt
package com.jetbrains.shared.FileStorage

import org.springframework.core.io.Resource
import org.springframework.web.multipart.MultipartFile

interface FileStorageService {
    fun storeFile(file: MultipartFile?): String

    fun downloadFile(fileName: String): Resource

    fun deleteFile(fileName: String)

    fun getFileExtension(fileName: String): String
}
